pkg/apis/core/validation/validation.go

```diff
@@ -5303,21 +5303,17 @@ func ValidateSecret(secret *core.Secret) field.ErrorList {
 			allErrs = append(allErrs, field.Invalid(dataPath.Key(core.DockerConfigJSONKey), "<secret contents redacted>", err.Error()))
 		}
 	case core.SecretTypeBasicAuth:
```

Comment:
the documentation for this secret type just indicates at least one of the fields must be set:

```
	// SecretTypeBasicAuth contains data needed for basic authentication.
	//
	// Required at least one of fields:
	// - Secret.Data["username"] - username used for authentication
	// - Secret.Data["password"] - password or token needed for authentication
	SecretTypeBasicAuth SecretType = "kubernetes.io/basic-auth"
```

we should not tighten validation here

Reply:
we should fix the documentation at https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret to be in line with the API doc and the existing validation

Reply:
Thanks for your comments, so can I just change this PR to reflect the documentation changes or should I close this and open a new one?

Reply:
I think the documentation change would need to be in a PR to https://github.com/kubernetes/website/

Reply:
A new PR to fix https://github.com/kubernetes/website/blob/main/content/en/docs/concepts/configuration/secret.md#basic-authentication-secret seems appropriate (and there might be other incorrect sections on that page).

Closing this since no code changes are recommended.

/close

Reply:
Sounds good, thanks for the feedback!